src/main/java/dev/openfeature/sdk/ImmutableTrackingEventDetails.java

```diff
@@ -0,0 +1,47 @@
+package dev.openfeature.sdk;
+
+import dev.openfeature.sdk.internal.ExcludeFromGeneratedCoverageReport;
+import lombok.Getter;
+import lombok.experimental.Delegate;
+
+import java.util.Map;
+import java.util.function.Function;
+
+
+/**
+ * ImmutableTrackingEventDetails represents data pertinent to a particular tracking event.
+ */
+public class ImmutableTrackingEventDetails implements TrackingEventDetails {
+
+    @Delegate(excludes = DelegateExclusions.class)
+    private final ImmutableStructure structure;
+
+    @Getter
+    private final Number value;
+
+    public ImmutableTrackingEventDetails() {
+        this.value = 0f;
```

Comment:
I think there `0` is not the right default. 
https://openfeature.dev/specification/sections/tracking#requirement-621 defines this as optional, so I think a numeric 0 is semantically wrong if no value is given. (same counts for `MutableTrackingEventDetails`)

Reply:
yeah good point, I've changed it to `null` and let the getter return an `Optional<Number>`

Reply:
That's a good solution. We already use optional and I think this conveys what we want.

Reply:
Nice that is a good solution!